### PR TITLE
Slime core removal surgery changes

### PR DESCRIPTION
**Pull request by MTandi**
## About The Pull Request

Fixed an issue with incorrect slime dead icon after core removal surgery. Made the surgery yield all cores in case if the slime had a steroid potion applied.

## Why It's Good For The Game

The icon was broken. Less clicks for surgery.

## Changelog

:cl:
fix: Fixed dead slime icon not showing when cores are extracted
qol: Slime core removal surgery extracts all cores on completion
/:cl:

### CODE REVIEW

**Comment by alien3301** [comment]
What's the point of upgrading the machine then?

**Comment by MTandi** [comment]
> What's the point of upgrading the machine then?

Faster processing

**Comment by nianjiilical** [comment]
maybe im reading this wrong but it feels like a pretty noticeable nerf to the slime grinder?? getting an earlier t3 update for 3 cores isnt hard at all, and you're dropping from (3-4 slimes x 2-4 cores =) 6-16 cores per adult slime to (3-4 slimes x 2 cores =) 6-8 cores per adult slime

its essentially removing the ability to upgrade slime core production in favor of making the slow, manual, entirely unused method of core extraction 'on par' with the grinder, by virtue of making the grinder worse

idk it just feels bad

**Comment by MTandi** [comment]
Probably going to preserve the original logic for processor and apply the core count to surgery only. This way you can extract 4 cores when you do surgery on adult slime even without t4 parts upgraded.

**Comment by carlarctg** [comment]
xenobio doesnt benefit from small tweaks and adjustments like this it just makes it even clunkier and annoying. leave it for the rework

**Comment by MTandi** [comment]
Made the surgery cores work independently from the processed cores. Technically you can now get up to 8 cores from a single slime.

**Comment by ExcessiveUseOfCobblestone** [comment]
Feel like we lost the plot on the surgery.

It is NOT supposed to be good, it is a way to extract a slime if power does not exist or you are processor-less, at the cost of being trash. Speed it up/take steps out to make it quick but dont touch its efficacy.

**Comment by MTandi** [comment]
Reduced the core count to 1 and 2 when doing surgery on baby and adult slime.

File: code/modules/food_and_drinks/machinery/processor.dm [hunk 1]
```diff
@@ -238,8 +238,8 @@
 		processed_slime.forceMove(drop_location())
 		processed_slime.balloon_alert_to_viewers("crawls free")
 		return
-	var/core_count = processed_slime.cores
-	for(var/i in 1 to (core_count+rating_amount-1))
+
+	for(var/i in 1 to rating_amount)
```
Comment (Ghommie) [inline code/modules/food_and_drinks/machinery/processor.dm:242]: You didn't fully revert the nerf smh.
```suggestion
	for(var/i in 1 to (processed_slime.cores+rating_amount-1))
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/machinery/processor.dm (hunk 1 above)
Comment (MTandi) [reply code/modules/food_and_drinks/machinery/processor.dm:242]: What nerf? 

The processing always gives 1 to 4 depending on the parts tier, regardless of the slime size.
The surgery gives 1 core from baby slime and 2 cores from adult slime.

Comment (Ghommie) [reply code/modules/food_and_drinks/machinery/processor.dm:242]: You aren't taking into account the slime steroid potion (which I'm not sure it's even useful) which increases the amount of cores a slime has.

**Comment by MTandi** [comment]
I forgot about the steroid potion. Removed core doubling on slime growth to avoid it having 10 cores when a 5-core baby grows up.

The PR is only about the icon fix and a surgery giving full amount of cores in case if the slime was on steroids.

**Pull request merged**